### ISSUE TITLE
db2 driver confuses schema name for database name

### ISSUE THREAD

**Issue by NattyNarwhal** [ADOdb/ADOdb#1032]
### Description

When trying to set the database name when connecting, the Db2 driver mixes up the DSN (containing the host information) and the database name. This is obviously invalid, and will fail to connect.

### Environment
- ADOdb version: 5.22.7
- Driver or Module: Db2
- Database type and version: Db2
- PHP version: 8.3.7
- Platform: IBM i 7.5

* [X] I have tested that the problem is reproducible in the [latest release](https://github.com/ADOdb/ADOdb/releases/latest) / [master](https://github.com/ADOdb/ADOdb/tree/master) branch / latest [hotfix](https://github.com/ADOdb/ADOdb/branches/all?query=hotfix) branch  
  ⚠️ Keep only the applicable option(s), i.e. where you actually tested, and remove the others

### Steps to reproduce

```php
$adoDbDir = "vendor/adodb/adodb-php";
require "$adoDbDir/adodb.inc.php";

$db = newAdoConnection("db2");
var_dump($db->connect("*LOCAL", "username", "password", "QIWS"));

$sql = "select * from qcustcdt"; // this table in QIWS
$array = $db->getAll($sql);

print_r($array);

```

Resulting error spew:

```
Warning: Because you provided user,password and database, DSN connection parameters were discarded
Catalogued connection using parameters: DB=QIWS / UID=username / PWD=password
PHP Warning:  db2_connect(): Connection failure userid (username) in /home/calvin/adodbtest/vendor/adodb/adodb-php/drivers/adodb-db2.inc.php on line 167
PHP Stack trace:
PHP   1. {main}() /home/calvin/adodbtest/index.php:0
PHP   2. ADOConnection->Connect($argHostname = '*LOCAL', $argUsername = 'username', $argPassword = 'password', $argDatabaseName = 'QIWS', $forceNew = *uninitialized*) /home/calvin/adodbtest/index.php:18
PHP   3. ADODB_db2->_connect($argDSN = '*LOCAL', $argUsername = 'username', $argPassword = 'password', $argDatabasename = 'QIWS') /home/calvin/adodbtest/vendor/adodb/adodb-php/adodb.inc.php:942
PHP   4. ADODB_db2->doDB2Connect($argDSN = '*LOCAL', $argUsername = 'username', $argPassword = 'password', $argDatabasename = 'QIWS', $persistent = *uninitialized*) /home/calvin/adodbtest/vendor/adodb/adodb-php/drivers/adodb-db2.inc.php:96
PHP   5. db2_connect($database = 'QIWS', $username = 'username', $password = 'password', $options = []) /home/calvin/adodbtest/vendor/adodb/adodb-php/drivers/adodb-db2.inc.php:167
PHP Warning:  db2_connect(): Relational database QIWS not in relational database directory. SQLCODE=-950 in /home/calvin/adodbtest/vendor/adodb/adodb-php/drivers/adodb-db2.inc.php on line 167
PHP Stack trace:
PHP   1. {main}() /home/calvin/adodbtest/index.php:0
PHP   2. ADOConnection->Connect($argHostname = '*LOCAL', $argUsername = 'username', $argPassword = 'password', $argDatabaseName = 'QIWS', $forceNew = *uninitialized*) /home/calvin/adodbtest/index.php:18
PHP   3. ADODB_db2->_connect($argDSN = '*LOCAL', $argUsername = 'username', $argPassword = 'password', $argDatabasename = 'QIWS') /home/calvin/adodbtest/vendor/adodb/adodb-php/adodb.inc.php:942
PHP   4. ADODB_db2->doDB2Connect($argDSN = '*LOCAL', $argUsername = 'username', $argPassword = 'password', $argDatabasename = 'QIWS', $persistent = *uninitialized*) /home/calvin/adodbtest/vendor/adodb/adodb-php/drivers/adodb-db2.inc.php:96
PHP   5. db2_connect($database = 'QIWS', $username = 'username', $password = 'password', $options = []) /home/calvin/adodbtest/vendor/adodb/adodb-php/drivers/adodb-db2.inc.php:167
*LOCAL: Relational database QIWS not in relational database directory. SQLCODE=-950
```

### Expected behavior

That it would connect to the specified system and set the default database.

### Additional context 

The old pre-rewrite version of the driver in 5.20 seemed to use the DSN intact and called `SET SCHEMA` after connecting: https://github.com/ADOdb/ADOdb/blob/4011379abe34845d44612dbe904f435b6cd81d9f/drivers/adodb-db2.inc.php#L72-L100

**Comment by mnewnham** [comment]
I'll take a look. I don't have a DB2 system myself at the moment either though.

**Comment by mnewnham** [comment]
probably a dupe of #1031. Please check the fix for that

**Comment by dregad** [comment]
Thanks for looking into it @mnewnham 

@NattyNarwhal please test Mark's proposed fix in #1034 and let us know if it fixes your problem.

**Comment by NattyNarwhal** [comment]
Using a full connection string doesn't work (tried with the PR). Note that there are [some IBM i specific things with those](https://github.com/php/doc-en/pull/3364); I've also never seen anyone use it like that.

```
LOG: Uncatalogued connection using DSN: DATABASE=QIWS;HOSTNAME=*LOCAL;UID=CALVIN;PWD=[...]
PHP Warning:  db2_connect(): Connection failure userid () in /home/calvin/adodbtest/vendor/adodb/adodb-php/drivers/adodb-db2.inc.php on line 179
PHP Warning:  db2_connect(): Error occurred in SQL Call Level Interface SQLCODE=-99999 in /home/calvin/adodbtest/vendor/adodb/adodb-php/drivers/adodb-db2.inc.php on line 179
```

I'll review the PR as well.

**Comment by dregad** [comment]
Fixed by #1034 (2bc630b08fa238eae697dd1fdbf89b6506e816be)
